.circleci/config.yml

```diff
@@ -74,6 +120,50 @@ jobs:
           name: Submit coverage data to codecov.
           command: bash <(curl -s https://codecov.io/bash)
           when: always
+      - run:
+          name: Print instructions on regeneration failure.
+          command: |
+            # Exit early if no diff was found
+            [[ ! -z ${DIFF_FOUND} ]] || exit 0
+            echo "Regenerating generated source probably caused the build to break, please run 'go run ./util/compile_protos' and fix the build."
+          when: on_fail
+
+  push-generated-source:
+    docker:
+      - image: golang:1.12
+    environment:
+      GO111MODULE: "on"
+    working_directory: /go/src/github.com/googleapis/gapic-showcase
+    steps:
+      - attach_workspace:
+          at: /go
+      - run:
+          name: Check if we are using generated source.
+          command: |
+            git status --porcelain
+            git diff --exit-code --quiet || echo 'export DIFF_FOUND="true"' >> $BASH_ENV
+      - run:
+          name: Create pull request if diff is found.
+          command: |
+            # Exit early if no diff was found
+            [[ ! -z ${DIFF_FOUND} ]] || exit 0
+            # Setup git config.
+            git config credential.helper 'cache --timeout=120'
+            git config user.email "googleapis-publisher@google.com"
+            git config user.name "googleapis-publisher"
+            # Push changes to branch.
+            export BRANCH="build-$(date +%Y%m%d%H%M%S)"
+            git checkout -b ${BRANCH}
+            git add .
+            git commit -m "Regenerate cmd, client, and server directories."
+            git push --quiet https://${GITHUB_TOKEN}@github.com/googleapis/gapic-showcase.git ${BRANCH}
+            # Create a PR with hub.
+            go get github.com/github/hub
+            hub pull-request \
```

Comment:
FYI this will use the the env var GITHUB_TOKEN to make this PR based on this documentation: https://hub.github.com/hub.1.html#github-oauth-authentication

Reply:
this is super tight